src/reducers/sources.js

```diff
@@ -377,6 +381,18 @@ function getSourcesByUrlInSources(
   return urls[url].map(id => sources[id]);
 }
 
+function getSourcesByUrlInSources5515(
```

Comment:
We shouldn't need this function; we should be able to do any structuring of the data in `getHasSiblingOfSameName`

Reply:
Do you mean all the changes should be done in the `src/components/PrimaryPanes/SourcesTreeItem.js`?

Reply:
Yes; it isn't apparent to me why we would need this extra function when we could do filtering in `getHasSiblingOfSameName`.  Also, why are we checking for `undefined`?